plugins/inputs/socket_listener/socket_listener.go

```diff
@@ -47,6 +48,27 @@ func (ssl *streamSocketListener) listen() {
 		}
 		ssl.connections[c.RemoteAddr().String()] = c
 		ssl.connectionsMtx.Unlock()
+
+		if ssl.KeepAlivePeriod != nil {
+			if tcpc, ok := c.(*net.TCPConn); ok {
+				if ssl.KeepAlivePeriod.Duration == 0 {
+					if err := tcpc.SetKeepAlive(false); err != nil {
+						ssl.AddError(fmt.Errorf("unable to disable keep alive: %s", err))
+					}
+				} else {
+					if err := tcpc.SetKeepAlive(true); err != nil {
+						ssl.AddError(fmt.Errorf("unable to enable keep alive: %s", err))
+					} else {
+						if err := tcpc.SetKeepAlivePeriod(ssl.KeepAlivePeriod.Duration); err != nil {
+							ssl.AddError(fmt.Errorf("unable to set keep alive period: %s", err))
```

Comment:
I think you could probably do something like `func setKeepAlive(*net.TCPConn, time.Duration) error`, might have to add the service address context to the error outside the function.  Would probably clean up the else branch when a duration is non-zero.  Let me know if you want to make changes or not.